### PR TITLE
fix(ListSeparator): add lineColor, textPadding, children, & textColor…

### PR DESCRIPTION
**Pull request by daanishnasir**

### CODE REVIEW

File: src/lib/List/index.js [hunk 1]
```diff
@@ -111,7 +117,11 @@ class List extends React.Component {
       []
     );
 
-    !isNaN(newIndex[0]) && this.setFocus(newIndex[0]);
+    children[newIndex[0]].type.displayName === 'ListSeparator' && !isNaN(newIndex[0])
+    ?
```
Comment (bfbiggs) [inline src/lib/List/index.js]: Same here - don't need a new new line for each of these.

File: src/lib/List/index.js [hunk 2]
```diff
@@ -111,7 +117,11 @@ class List extends React.Component {
       []
     );
 
-    !isNaN(newIndex[0]) && this.setFocus(newIndex[0]);
```
Comment (bfbiggs) [inline src/lib/List/index.js]: Why was this setFocus code removed? I would suggest looking at git blame and then check the component it was added to address.

File: src/lib/ListSeparator/index.js [hunk 3]
```diff
@@ -8,24 +8,61 @@ import PropTypes from 'prop-types';
  */
 
 const ListSeparator = props => {
-  const { text } = props;
+  const {
+    children,
+    className,
+    textColor,
+    lineColor,
```
Comment (bfbiggs) [inline src/lib/ListSeparator/index.js]: alphabetize props

File: src/lib/ListSeparator/index.js [hunk 4]
```diff
@@ -8,24 +8,61 @@ import PropTypes from 'prop-types';
...
-        `cui-list-separator` + 
-        `${text ? ' cui-list-separator--text' : ''}`
-      }>
-      {text && <p className='cui-list-separator__text'>{text}</p>}
+        `cui-list-separator` +
+        `${text || children ? ' cui-list-separator--text' : ''}` +
+        `${(className && ` ${className}`) || ''}`
+      }
+      style={{
+        backgroundColor:!text && lineColor && lineColor,
```
Comment (bfbiggs) [inline src/lib/ListSeparator/index.js]: So - if they add text - they can't change the color?

File: src/lib/ListSeparator/index.js [hunk 5]
```diff
@@ -8,24 +8,61 @@ import PropTypes from 'prop-types';
...
+        margin:text && '0',
```
Comment (bfbiggs) [inline src/lib/ListSeparator/index.js]: Shouldn't the default be 0 and consumers add margin as needed? The Asana task had a criteria to remove fixed margins.

File: src/lib/ListSeparator/index.js [hunk 6]
```diff
@@ -8,24 +8,61 @@ import PropTypes from 'prop-types';
...
+      {...otherProps}
+    >
+
+      {
+        children || text
+        ?
+          <span
+            className='cui-list-separator__text'
+            style={{
+              color: textColor || '',
```
Comment (bfbiggs) [inline src/lib/ListSeparator/index.js]: I think we would want to  make this property conditional if it doesn't exist instead of a blank string.

**Comment by bfbiggs** [comment]
Please provide a screenshot with a visual PR.

This is what I'm seeing below when I build it with the core PR.  @pauljeter have you built it locally?
screen shot 2018-08-21 at 12 06 14 pm

File: src/lib/List/index.js [hunk 7]
```diff
@@ -94,10 +94,16 @@ class List extends React.Component {
           ? Math.abs(currentIdx + idx - length)
           : currentIdx + idx + 1;
 
-        const label =
+          const label =
```
[the remark this answers is not in the document]
Comment (daanishnasir) [reply src/lib/List/index.js]: label tabbed fix, i typically indent the return in ternary's, let me know if that style should change. 
screen shot 2018-08-21 at 12 46 22 pm

[the remark this answers is not in the document]
Comment (daanishnasir) [reply src/lib/List/index.js]: fixed

File: src/lib/List/index.js (hunk 1 above)
Comment (daanishnasir) [reply src/lib/List/index.js]: fixed to same line as return

File: src/lib/ListSeparator/index.js (hunk 3 above)
Comment (daanishnasir) [reply src/lib/ListSeparator/index.js]: alphabetized.

File: src/lib/ListSeparator/index.js (hunk 6 above)
Comment (daanishnasir) [reply src/lib/ListSeparator/index.js]: changed to conditional; textColor && textColor

File: src/lib/List/index.js (hunk 2 above)
Comment (daanishnasir) [reply src/lib/List/index.js]: Checked other components that are using <List> to make sure functionality still works. setFocus moved down a few lines inside of ternary

File: src/lib/ListSeparator/index.js (hunk 4 above)
Comment (daanishnasir) [reply src/lib/ListSeparator/index.js]: fixed, they can change color of line when they add text through css variables.

File: src/lib/List/index.js [hunk 8]
```diff
@@ -95,9 +95,11 @@ class List extends React.Component {
           : currentIdx + idx + 1;
 
         const label =
-          arr[index].props.role === 'listItem' || arr[index].type.displayName === 'SelectOption'
-            ? arr[index].props.label
-            : arr[index].props.header;
+          arr[index].type.displayName === "ListSeparator"
+          ? arr[index+1].props.label || arr[index+1].props.header
```
Comment (bfbiggs) [inline src/lib/List/index.js]: tabbing needs to indicated the nested ternary:
```
       arr[index].type.displayName === "ListSeparator"
          ? arr[index+1].props.label || arr[index+1].props.header
          : arr[index].props.role === 'listItem' || arr[index].type.displayName === 'SelectOption'
            ? arr[index].props.label
            : arr[index].props.header;
```

File: src/lib/ListSeparator/index.js [hunk 9]
```diff
@@ -8,24 +8,63 @@ import PropTypes from 'prop-types';
...
-        `cui-list-separator` + 
-        `${text ? ' cui-list-separator--text' : ''}`
-      }>
-      {text && <p className='cui-list-separator__text'>{text}</p>}
+        'cui-list-separator' +
+        `${className && ` ${className}` || ''}`
+      }
+      style={{
+        color: lineColor ? lineColor : '#EBEBEB',
+        margin: margin && `${margin}px 0`
```
Comment (bfbiggs) [inline src/lib/ListSeparator/index.js]: we need to treat padding and margin the same so just allow a string and let it be passed in without the 0's or the px's

**Pull request merged**